clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -588,13 +669,6 @@ private void updateLastAckedOffset(ProduceResponse.PartitionResponse response, P
     }
 
     public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceResponse.PartitionResponse response) {
-        if (!hasProducerIdAndEpoch(batch.producerId(), batch.producerEpoch())) {
-            log.debug("Ignoring completed batch {} with producer id {}, epoch {}, and sequence number {} " +
-                            "since the producerId has been reset internally", batch, batch.producerId(),
-                    batch.producerEpoch(), batch.baseSequence());
-            return;
-        }
-
         maybeUpdateLastAckedSequence(batch.topicPartition, batch.baseSequence() + batch.recordCount - 1);
```

Comment:
Another place where a `lastSequence` method would be useful.

Reply:
Added one to ProducerBatch that computes based on record count